modules/persistence/src/services/pages/index.ts

```diff
@@ -184,24 +184,42 @@ const updatePages = async (pages: Document[], collection: string) => {
     return;
   }
 
+  const timerLabel = 'page document updates';
+  console.time(timerLabel);
+
   // Find all pages that share the same project name + branch. Expects page IDs
   // to include these two properties after parse
   const pageIdPrefix = pages[0].page_id.split('/').slice(0, 3).join('/');
   const previousPagesCursor = await findPrevPageDocs(pageIdPrefix, collection);
   const { mapping: prevPageDocsMapping, pageIds: prevPageIds } = await createPageAstMapping(previousPagesCursor);
 
+  const diffsTimerLabel = 'finding page differences';
+  console.time(diffsTimerLabel);
   const updatedPagesManager = new UpdatedPagesManager(prevPageDocsMapping, prevPageIds, pages);
   const operations = updatedPagesManager.getOperations();
+  console.timeEnd(diffsTimerLabel);
 
   if (operations.length > 0) {
+    const bulkWriteTimerLabel = 'page document update writes';
+    console.time(bulkWriteTimerLabel);
     await bulkWrite(operations, collection);
```

Comment:
similar here. can we wrap these in a try/catch/finally block to catch the timer? seems like errors from bulkWrite will also be uncaught

Reply:
Wrapped in a try/finally block. Ended up wrapping the main body of the `updatePages` function in a try/catch/finally to catch errors at a function-wide level